src/app/backend/resource/dataselect/dataselect.go

```diff
@@ -107,16 +107,11 @@ func (self *DataSelector) Filter() *DataSelector {
 	filteredList := []DataCell{}
 
 	for _, c := range self.GenericDataList {
-		matches := true
 		for _, filterBy := range self.DataSelectQuery.FilterQuery.FilterByList {
 			v := c.GetProperty(filterBy.Property)
-			if v == nil || !v.Contains(filterBy.Value) {
-				matches = false
-				break
-			}
-		}
-		if matches {
-			filteredList = append(filteredList, c)
+			if v == nil || v.Contains(filterBy.Value) {
```

Comment:
Can you explain this change?

Reply:
I just realized a mistake in this, should've removed the `v==nil`

Reply:
So overall
Currently this function Filter(), basically takes the intersection of various `filterBy` conditions passed to it

I instead changed it to work in a union fashion

Because I wanted to get results filtered by either name or labels whichever is found

I even mentioned this decision on slack.